Adding new faction retrieval function

raecyst opened the pull request:
Faction retrieval function to save space as well as provide a much quicker method of retrieving a faction.

Can easily save so much space in files.

raecyst committed:
Adding new faction retrieval function

Faction retrieval function to save space as well as provide a much quicker method of retrieving a faction.

raecyst committed:
Fixed double error

Fixed a double error with the new function.

TovarischPootis commented:
while at the core the change is minor in terms of existing framework code. It could very much be a useful tool for custom commands and such that require the faction name (plytransfer/whitelist for example). Convenience if nothing else

raecyst committed:
Fixing an indexing nil error

An error can occur when you first join due to the trace being invalid. Added a check that prevents that occuring.

its-johnny commented:
Yeah why not
